graylog2-server/src/main/java/org/graylog2/shared/rest/documentation/generator/Generator.java

```diff
@@ -81,18 +85,37 @@
     private static Map<String, Object> overviewResult = Maps.newHashMap();
     private static Reflections reflections;
 
+    private final Map<String, String> packageNames;
     private final ObjectMapper mapper;
 
-    public Generator(String[] packageNames, ObjectMapper mapper) {
+    public Generator(Map<String, String> packageNames, ObjectMapper mapper) {
+        this.packageNames = packageNames;
         this.mapper = mapper;
 
         if (reflections == null) {
-            reflections = new Reflections((Object[]) packageNames);
+            reflections = new Reflections((Object[]) packageNames.keySet().toArray());
         }
     }
 
     public Generator(String packageName, ObjectMapper mapper) {
-        this(new String[]{packageName}, mapper);
+        this(ImmutableMap.of(packageName, ""), mapper);
+    }
+
+    public String prefixedPath(Class<?> resourceClass, @Nullable String annotationPath) {
+        final String path = (annotationPath == null ? "" : annotationPath);
+        final String resourcePackage = resourceClass.getPackage().getName();
+        final Optional<String> key = packageNames.keySet().stream().filter(resourcePackage::startsWith).findFirst();
```

Comment:
It was not immediately clear to me, why this was needed or what the contents of this variable will be. Maybe `parentPackage` would be a better name?
Additionally, `packageNames.keySet()` isn't necessarily ordered or correctly ordered. Does this matter at all?